src/main/java/org/terasology/HealthComponent.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.network.Replicate;
+import org.terasology.rendering.nui.properties.TextField;
+
+public class HealthComponent implements Component {
+
+    @Replicate
+    public int maxHealth = 20;
+    @Replicate
+    public float regenRate;
+    @Replicate
+    public float waitBeforeRegen;
+
+    @Replicate
+    public float fallingDamageSpeedThreshold = 20;
+    @Replicate
+    public float horizontalDamageSpeedThreshold = 20;
+    @Replicate
+    public float excessSpeedDamageMultiplier = 10f;
+
+    @Replicate
+    @TextField
+    public int currentHealth = 20;
+
+    // Regen info
+    public long nextRegenTick;
+
+    public boolean destroyEntityOnNoHealth;
+
+    public HealthComponent() {
+    }
+
+    public HealthComponent(int maxHealth, float regenRate, float waitBeforeRegen) {
```

Comment:
I figure this constructor is here because it already existed - I wonder how frequently it is used and if it is really justified for it to exist? It might be a good practice for us to always encourage the pattern of setting whichever exact fields you need rather than use a custom constructor

```java
HealthComponent health;
health.maxHealth = 123;
health.regenRate = 42;
//apply the component to something
```

Reply:
After searching in the mega workspace, I couldn't find any usage of this constructor. I am hence removing this constructor.
And, since the unique constructor is removed, the default constructor is not needed explicitly.